### PR TITLE
Clean Software Code

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Work done before the UI migrations related to Software.

### CODE REVIEW

File: src/Item_SoftwareLicense.php [hunk 1]
```diff
@@ -537,19 +526,10 @@ public static function showForLicense(SoftwareLicense $license)
...
-            $start = 0;
-        }
-
-        if (isset($_GET["order"]) && ($_GET["order"] == "DESC")) {
-            $order = "DESC";
-        } else {
-            $order = "ASC";
-        }
+        $start = (int) ($_GET["start"] ?? 0);
+        $order = $_GET['order'] === 'DESC' ? 'DESC' : 'ASC';
```
Comment (trasher) [inline src/Item_SoftwareLicense.php:530]: undefined case is not handled

File: src/Item_SoftwareVersion.php [hunk 2]
```diff
@@ -1009,14 +968,13 @@ public static function getFromItem(CommonDBTM $item, $sort = null, $order = null
...
-        $iterator = $DB->request($request);
-        return $iterator;
+        return $DB->request($request);
     }
 
     /**
      * Show software installed on a computer
      *
-     * @param Computer $comp         Computer object
+     * @param CommonDBTM $item         Computer object
```
Comment (trasher) [inline src/Item_SoftwareVersion.php:977]:
```suggestion
     * @param CommonDBTM $item         Item
```

File: src/Software.php [hunk 3]
```diff
@@ -212,17 +191,11 @@ public static function updateValidityIndicator($ID)
...
      * Print the software form
      *
-     * @param $ID integer ID of the item
-     * @param $options array
-     *     - target filename : where to go when done.
-     *     - withtemplate boolean : template or basic item
-     *
-     * @return boolean item found
-     **/
+     * {@inheritdoc}
```
Comment (trasher) [inline src/Software.php:197]: This whole comment may be removed

File: src/SoftwareLicense.php [hunk 4]
```diff
@@ -979,19 +920,10 @@ public static function showForSoftware(Software $software)
...
-            $start = 0;
-        }
-
-        if (isset($_GET["order"]) && ($_GET["order"] == "DESC")) {
-            $order = "DESC";
-        } else {
-            $order = "ASC";
-        }
+        $start = (int) ($_GET['start'] ?? 0);
+        $order = $_GET['order'] === 'DESC' ? 'DESC' : 'ASC';
```
Comment (trasher) [inline src/SoftwareLicense.php:924]: undefined case not handled

**Comment by trasher** [comment]
Also, tests are failing :/

**Review by trasher — APPROVED** [review]

File: src/SoftwareLicense.php [hunk 5]
```diff
@@ -826,17 +772,17 @@ public static function cronSoftware($task = null)
...
                 if (NotificationEvent::raiseEvent('alert', new self(), $options)) {
-                    $entityname = Dropdown::getDropdownName("glpi_entities", $entity);
+                    $entityname = Dropdown::getDropdownName(Entity::getTable(), $entity);
                     if ($task) {
                         //TRANS: %1$s is the entity, %2$s is the message
                         $task->log(sprintf(__('%1$s: %2$s') . "\n", $entityname, $message));
                         $task->addVolume(1);
                     } else {
-                        Session::addMessageAfterRedirect(sprintf(
+                        Session::addMessageAfterRedirect(htmlspecialchars(sprintf(
```
Comment (cedric-anne) [inline src/SoftwareLicense.php:781]: `$message` contains a `<br>` tag for each licence. These `<br>` tags should not be encoded.

File: src/SoftwareLicense.php [hunk 6]
```diff
@@ -979,19 +920,10 @@ public static function showForSoftware(Software $software)
...
-            $start = 0;
-        }
-
-        if (isset($_GET["order"]) && ($_GET["order"] == "DESC")) {
-            $order = "DESC";
-        } else {
-            $order = "ASC";
-        }
+        $start = (int) ($_GET['start'] ?? 0);
+        $order = ($_GET['order'] ?? 'DESC') === 'DESC' ? 'DESC' : 'ASC';
```
Comment (cedric-anne) [inline src/SoftwareLicense.php:924]:
```suggestion
        $order = ($_GET['order'] ?? 'ASC') === 'DESC' ? 'DESC' : 'ASC';
```

File: src/SoftwareLicense.php [hunk 7]
```diff
@@ -1258,45 +1185,40 @@ public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
                     return self::createTabEntry(
                         self::getTypeName(Session::getPluralNumber()),
                         (($nb >= 0) ? $nb : '&infin;'),
-                        $item::getType()
+                        $item::class
                     );
 
-                case SoftwareLicense::class:
+                case __CLASS__:
```
Comment (cedric-anne) [inline src/SoftwareLicense.php:1191]: We use the `::class` notation everywhre. It seems more logical to use `self::class` here.
```suggestion
                case self::class:
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**